Add new tests for preference panels.

pabender opened the pull request.

rhwood commented on java/test/jmri/swing/PreferencesPanelTestBase.java:
```diff
@@ -0,0 +1,99 @@
+ *
+ * @author Paul Bender Colyright (C) 2020
+ */
+abstract public class PreferencesPanelTestBase {
```
This class should use a generic `PreferencesPanelTestBase<P extends PreferencesPanel>` to avoid any potential need to cast `prefsPanel` to the tested class in the actual test classes.

pabender replied:
thanks. will fix

rhwood commented on java/test/apps/FileLocationPaneTest.java:
```diff
@@ -1,33 +1,41 @@
+    @Override
+    @Test
+    public void getPreferencesTooltip() {
+        // should this actually return null?
```
We have not stipulated that all preferences tabs in a tabbed pane require a tooltip, so, yes, those classes that return null for the preferences tooltip should return null (the method that cares is `javax.swing.JTabbedPane.insertTab(...)` and it accepts null for a tooltip).

pabender replied:
Thanks.  Will Fix

rhwood commented on java/src/jmri/jmrit/consisttool/ConsistTool.properties:
```diff
@@ -53,3 +53,4 @@ ScanConsists = Scan Roster for Consists
 WriteCVToRosterOption = Update CV19 in roster when modifying advanced consists
 WriteCVToRosterToolTip = When modifying advanced consists, CV19 is changed.  Selecting this option to update the roster entry, when known, when CV19 is updated.
 StartupConsistToolAction=Open Consisting Tool
+ConsistPreferencesToolTip = Preferences for the consist tool.
```
Tooltips should only be used when they can provide *more* information, and a tooltip displayed in the Preferences window over "Consist Control" that reads "Preferences for the consist tool" doesn't. I would remove this.

pabender replied:
thanks, will fix

rhwood commented on java/test/apps/ManagerDefaultsConfigPaneTest.java:
```diff
@@ -1,35 +1,39 @@
+    @Override
+    @Test
+    public void getPreferencesTooltip() {
+        // should this actually return null?
```
Yes it should.

rhwood commented on java/test/apps/SystemConsoleConfigPanelTest.java:
```diff
@@ -1,33 +1,38 @@
+    @Override
+    @Test
+    public void getPreferencesTooltip() {
+        // should this actually return null?
```
Yes, it should.

rhwood commented on java/test/apps/startup/StartupActionsPreferencesPanelTest.java:
```diff
@@ -1,32 +1,52 @@
+    @Override
+    @Test
+    public void getPreferencesTooltip(){
+        // This class returns null for preferences tool tip, but should it?
```
Yes, it should.

pabender replied:
thanks. will fix.

rhwood commented on java/test/apps/startup/StartupActionsPreferencesPanelTest.java:
```diff
@@ -1,32 +1,52 @@
-        JUnitUtil.tearDown();
+    @Override
+    @Test
+    public void getTabbedPreferencesTitle(){
```
Since a preferences panel controls if it is put into a tab with another preference panel or not, you should expect that preferences panels that are not under tabs will return null for the title for the tab.

pabender replied:
thanks.  will fix.

rhwood commented on java/test/jmri/jmris/json/JsonServerPreferencesPanelTest.java:
```diff
@@ -11,23 +12,21 @@
+    @Override
+    @Test
+    public void getLabelKey(){
+        // This class returns null for label key, but should it?
```
The only preferences panels that should not be returning null label keys are preferences panels that are tabbed together with other preferences panels. Panels showing up in the list on the preferences window should be returning null for label keys.

pabender replied:
Thanks, will fix

rhwood commented on java/test/jmri/swing/PreferencesPanelTestBase.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    @Test
+    public void getPreferencesTooltip() {
```
Since the PreferencesPanel interface stipulates no requirement of nullness, this test should not assert is not null. If it is to assert anything, it should assert there is no exception.

pabender replied:
thanks, will fix

rhwood commented on java/test/jmri/swing/PreferencesPanelTestBase.java:
```diff
@@ -0,0 +1,99 @@
+
+    @Test
+    public void getLabelKey() {
+        assertThat(prefsPanel.getLabelKey()).isNotNull();
```
Since the PreferencesPanel interface stipulates no requirement of nullness, this test should not assert is not null. If it is to assert anything, it should assert there is no exception.

pabender replied:
thanks, will fix

bobjacobsen commented:
Is there a preference between 
```
    @Disabled
    @ToDo("fix setup so the parent class test can run")
```
and
```
    @Disabled("fix setup so the parent class test can run")
```
under JUnit5 for should phrases like that?  If so, I'll add it to our JUnit page.  Thanks.

pabender commented:
> Is there a preference between
> 
> ```
>     @Disabled
>     @ToDo("fix setup so the parent class test can run")
> ```
> 
> and
> 
> ```
>     @Disabled("fix setup so the parent class test can run")
> ```
> 
> under JUnit5 for should phrases like that? If so, I'll add it to our JUnit page. Thanks.

They really should both have notes.  I think I fixed the `@Disabled` annotations  that had an missing note.

The `@Disabled` is the JUnit 5 version of `@Ignore` And our rules require the reason for the ignore.  The `@ToDo` Is our required note when an empty test is present so the parent class test doesn’t run, but the parent class is applicable to the class under test.

I updated the documentation for JUnit 5 in PR #8855.

pabender commented:
Travis headless restarted after

[INFO] 
9407[INFO] Results:
9408[INFO] 
9409[ERROR] Errors: 
9410[ERROR]   LoaderPaneTest.testLength:62 » ConcurrentModification
9411[INFO] 
9412[ERROR] Tests run: 27145, Failures: 0, Errors: 1, Skipped: 3939
9413

bobjacobsen commented:
> They really should both have notes. I think I fixed the @Disabled annotations that had an missing note.
> 
> The @Disabled is the JUnit 5 version of @Ignore And our rules require the reason for the ignore. The @ToDo Is our required note when an empty test is present so the parent class test doesn’t run, but the parent class is applicable to the class under test.
> 
> I updated the documentation for JUnit 5 in PR #8855.

Thank you!
